src/build/arg/mod.rs

```diff
@@ -5024,6 +5031,9 @@ impl<'help> Display for Arg<'help> {
         if self.is_set(ArgSettings::TakesValue) || self.is_positional() {
             display_arg_val(self, |s, _| write!(f, "{}", s))?;
         }
+        if !self.is_positional() && self.is_set(ArgSettings::TakesValue) && is_optional_val {
```

Comment:
How about we instead have a `needs_closing_bracket` variable that gets set to true with `is_optional_val` and we base the close off of it.  That way we don't have to worry about this staying correct as the code evolves.

Reply:
Great suggestion. Will do.